apm-agent-core/src/main/java/co/elastic/apm/agent/impl/ActiveStack.java

```diff
@@ -122,22 +122,15 @@ boolean deactivate(ElasticContext<?> context, List<ActivationListener> activatio
         ElasticContext<?> activeContext = currentContext();
         activeContextStack.remove();
 
-        AbstractSpan<?> span = context.getSpan();
-
-        if (activeContext != context && activeContext instanceof ElasticContextWrapper) {
```

Comment:
This unwrapping was unnecessary: `currentContext` has already been doing the unwrapping.